examples/scene/bsn.rs

```diff
@@ -0,0 +1,197 @@
+//! This is a temporary stress test of various bsn! features.
+// TODO: move these into actual tests and replace this with a more instructive user-facing example
+use bevy::{
+    prelude::*,
+    scene2::prelude::{Scene, *},
+};
+use bevy_scene2::SceneList;
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_systems(Startup, setup)
+        .add_systems(Update, (print::<Sprite>, print::<Transform>, print::<Team>))
+        .run();
+}
+
+fn setup(mut commands: Commands, assets: Res<AssetServer>) {
+    assets.load_scene("scene://base.bsn", base());
+    assets.load_scene("scene://transform_1000.bsn", transform_1000());
+    let top_level_handle = assets.load_scene("scene://top_level.bsn", top_level());
+    commands.spawn(ScenePatchInstance(top_level_handle));
+}
+
+fn top_level() -> impl Scene {
+    let a = 20usize;
+    let b = 1993usize;
+    bsn! {
+        #TopLevel
+        :"scene://base.bsn"
+        :x
+        Sprite { size: b }
+        Team::Green(10)
+        {transform_1337()}
+        Children [
+            Sprite { size: {4 + a}}
+        ]
+    }
+}
+
+fn base() -> impl Scene {
+    let sprites = (0..10usize)
+        .map(|i| bsn! {Sprite { size: {i} }})
+        .collect::<Vec<_>>();
+
+    bsn! {
+        Name("Base")
+        Sprite {
+            handle: "asset://branding/bevy_bird_dark.png",
+            size: 1,
+            nested: Nested {
+                handle: @"asset://hello.png"
+            }
+        }
+        Transform::from_translation(Vec3::new(1.0, 1.0, 1.0))
+        Team::Red {
+            x: 10,
+            y: Nested {
+                foo: 10
+            },
+        }
+        Gen::<usize> {
```

Comment:
Have you considered allowing omitting the :: in the turbofish as a syntax sugar? Not sure if it can be resolved in all cases but it seems like it should be possible